raiden/channel/balance_proof.py

```diff
@@ -209,30 +209,38 @@ def get_known_unlocks(self):
             self.hashlock_unlockedlocks.itervalues(),
         )
 
+        tree = self.generate_merkle_tree();
+
         return [
             self.compute_proof_for_lock(
                 partialproof.secret,
                 partialproof.lock,
+                tree,
             )
             for partialproof in allpartialproof
         ]
 
-    def compute_proof_for_lock(self, secret, lock):
-        alllocks = chain(
-            self.hashlock_pendinglocks.values(),
-            self.hashlock_unclaimedlocks.values(),
-            self.hashlock_unlockedlocks.values()
-        )
+    def compute_proof_for_lock(self, secret, lock, tree=None):
```

Comment:
I'm happy to do it that way, I went for the optional argument approach as it is used in other functions in that source (register_secret and release_lock_by_secret) as well as in other files across the codebase but if you'd prefer more-but-simpler functions I can rewrite and resubmit.